plasmapy/analysis/swept_langmuir/ion_saturation_current.py

```diff
@@ -0,0 +1,228 @@
+"""
+Functionality for determining the ion-saturation current of a Langmuir sweep.
+"""
+__all__ = ["find_ion_saturation_current", "ISatExtras"]
+__aliases__ = ["find_isat_"]
+
+import numbers
+import numpy as np
+
+from typing import Tuple, NamedTuple, Union
+
+from plasmapy.analysis import fit_functions as ffuncs
+from plasmapy.analysis.swept_langmuir.helpers import check_sweep
+
+__all__ += __aliases__
+
+
+class ISatExtras(NamedTuple):
+    """
+    Create a tuple containing the extra parameters calculated by
+    `find_ion_saturation_current`.
+    """
+    rsq: Union[float, None]
+    """
+    Alias for field number 0, the r-squared value of the ion-saturation
+    curve fit.
+    """
+
+    fitted_func:  Union[ffuncs.AbstractFitFunction, None]
+    """
+    Alias for field number 1, the :term:`fit-function` fitted during
+    the ion-saturation curve fit.
+    """
+
+    fitted_indices: Union[slice, None]
+    """
+    Alias for field number 2, the indices used in the ion-saturation
+    curve fit.
+    """
+
+
+def find_ion_saturation_current(
+    voltage: np.ndarray,
+    current: np.ndarray,
+    *,
+    fit_type: str = "exp_plus_linear",
+    current_bound: float = None,
+    voltage_bound: float = None,
+) -> Tuple[ffuncs.Linear, ISatExtras]:
+    """
+    Determines the ion-saturation current (:math:`I_{sat}`) for a given
```

Comment:
[optional] I'm wondering if ```:math:`I_{sat}` ``` is used often enough to warrant a reST substitution in `docs/common_links.rst` (possibly with the `\` at the end to help with spacing).

Reply:
At this point I don't thinks so, but might I might change my tune as the `swept_langmuir` module expands.